### PR TITLE
[SYCLomatic] Report warning if there is zero-length array used in device code

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/test/dpct/zero_length_array.cu [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// CHECK-NEXT:   DPCT1102:{{[0-9]+}}: Zero-length arrays are not permitted in SYCL device code.
+// CHECK-NEXT:   */
+// CHECK-NEXT:   int arr0[0];
+// CHECK-NEXT:   int arr1[1];
+// CHECK-NEXT:   // zero-sized local_accessor is allowed in SYCL
+// CHECK-NEXT: }
+__global__ void k() {
+  int arr0[0];
+  int arr1[1];
+  __shared__ int arr2[0];// zero-sized local_accessor is allowed in SYCL
```
Comment (tomflinda) [inline clang/test/dpct/zero_length_array.cu:17]: Comment should be updated to "// zero-sized local_accessor is not allowed in SYCL"

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 2]
```diff
@@ -2080,6 +2080,36 @@ void AtomicFunctionRule::runRule(const MatchFinder::MatchResult &Result) {
 
 REGISTER_RULE(AtomicFunctionRule, PassKind::PK_Migration)
 
+void ZeroLengthArrayDetectorRule::registerMatcher(MatchFinder &MF) {
+  MF.addMatcher(typeLoc(loc(constantArrayType())).bind("ConstantArrayType"),
+                this);
+}
+void ZeroLengthArrayDetectorRule::runRule(
+    const MatchFinder::MatchResult &Result) {
```
Comment (zhimingwang36) [inline clang/lib/DPCT/ASTTraversal.cpp:2088]: Please add comment of the rule: 1. emit warnning when zero length array in device code and it is not share local memroy.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**